New unit tests for connect-datadog library

rue-mchang opened the pull request:
* Add robustness by adding unit tests to the repo.
* Introduce the jest library for tests

rue-mchang committed:
New unit tests for connect-datadog library, full coverage

nsheaps committed:
Add tracking for request method type:

- Configurable with 'method' option on middleware initialization

rue-dhurd commented:
Let's clean up these commit messages. Capitalize the first letter, and squash the two commits together.

Do these unit tests cover _everything_? If not, lets call out what parts of the middleware they do cover in the commit message.

nsheaps committed:
Remove duplicate `method` handling

It already existed so adding the code was not the appropriate path to take. In the mean time, I standardized the `method` option to be treated similarly to how the other options are treated

nsheaps committed:
Re-add mistakenly removed test

nsheaps committed:
Remove unnecessary setup on req.method

That’s now part of test setup for the request object

rue-mchang committed:
Enable base_url in tags, base_url tests, handle passed options consistently
